likwid: Fix the perms script

christian-heusel opened the pull request:
The script loops over the path (which includes the prefix), but
additionally adds the prefix up front which results in duplicate paths
and a double "/" in the command like in the following one:

    chown root:root /opt/csg/spack/opt/spack/linux-debian12-zen2/gcc-12.2.0/likwid-5.4.1-xfc6quebnf2kosydl3ospaeoskxnxwhn//opt/csg/spack/opt/spack/linux-debian12-zen2/gcc-12.2.0/likwid-5.4.1-xfc6quebnf2kosydl3ospaeoskxnxwhn/sbin/likwid-accessD

Additionally the path is currently not quoted which can potentially
result in word splitting for weird paths.

Signed-off-by: Christian Heusel <christian@heusel.eu>

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

christian-heusel committed:
likwid: Fix the perms script

The script loops over the path (which includes the prefix), but
additionally adds the prefix up front which results in duplicate paths
and a double "/" in the command like in the following one:

    chown root:root /opt/csg/spack/opt/spack/linux-debian12-zen2/gcc-12.2.0/likwid-5.4.1-xfc6quebnf2kosydl3ospaeoskxnxwhn//opt/csg/spack/opt/spack/linux-debian12-zen2/gcc-12.2.0/likwid-5.4.1-xfc6quebnf2kosydl3ospaeoskxnxwhn/sbin/likwid-accessD

Additionally the path is currently not quoted which can potentially
result in word splitting for weird paths.

Signed-off-by: Christian Heusel <christian@heusel.eu>

christian-heusel commented:
Also is this file in the right place? 🤔 Having it installed to the root folder of the spack directory structure for `likwid` feels weird to me 😅

TomTheBear commented:
Looks good.

I don't know where it gets installed and why. If you know a more suitable directory, feel free to change it.

christian-heusel commented:
This is the current directory structure, the script just gets installed into the root folder (which is wrong IMO):
```
/opt/csg/spack/opt/spack/linux-debian12-ivybridge/gcc-12.2.0/likwid-5.4.1-pvzipg4lfihkdrqivc2t5hhpnfc3vxvs/
├── bin
├── include
├── lib
├── man
├── sbin
├── share
└── spack_perms_fix.sh
```

TomTheBear commented:
The file location seems fine to me. It could stay in the temporary build folder but my guess is that the folder gets deleted after successful installation. The root directory after installation is ok for me. The path is not made available later (unlike bin, sbin, lib, include and man). Maybe the .spack folder, it should be there as well.

The permissions cannot be fixed in the package.py . Commonly, spack runs with user-permissions but the installation with accessdaemon requires root. That's why it is offloaded to this permissions script to run it after installation with proper permissions.

TomTheBear commented:
Did you use https://github.com/RRZE-HPC/likwid/tree/master/packaging/ansible ? It might be outdated. If your ansible script is general enough to be used by others, feel free to add it to likwid.

I agree, this PR can be merged.

TomTheBear reviewed (approved):
Looks good, ready to go.

wdconinc commented on var/spack/repos/builtin/packages/likwid/spack_perms_fix.sh.j2:
We should use a different name, since there are other packages (singularity, apptainer) that also use this exact filename (also problematic on their end!), and it becomes impossible to tell which script belongs to which package. Suggestion: `spack_perms_fix_likwid.sh`. Then install into `bin` per filesystem hierarchy standard, not the prefix root.

christian-heusel replied:
I have now added a second commit that renames & moves the script, please re-review if that is something like you wanted it to be. I can also submit similar patches for apptainer & singularity once that one here has been approved 😊

wdconinc replied:
It is a bit trickier for apptainer and singularity because the file is handled in a base class. I don't know whether it knows the correct name at that level, or it will just return some generic name. #49102 was my attempt until I realized that this wasn't as trivial. Will need more testing.

christian-heusel replied:
Ah if you're already on it I guess I'll leave it to the experts 😜

wdconinc reviewed (commented):
Sorry for delayed review.

> I don't know where it gets installed and why. If you know a more suitable directory, feel free to change it.

I made some comments in this regard.

christian-heusel committed:
likwid: Make the perm scripts' name unique

Also move it into the proper binary folder as per the Filesystem
Hierarchy Standard.

Signed-off-by: Christian Heusel <christian@heusel.eu>

wdconinc reviewed (approved):
Thanks. Looks good to me!
